sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -150,6 +153,14 @@ class DataFrame private[sql](
     }
   }
 
+  private[sql] def resolveToIndex(colName: String): Option[Int] = {
+    val resolver = sqlContext.analyzer.resolver
+    // First remove any user supplied quotes.
+    val unquotedColName = colName.stripPrefix("`").stripSuffix("`")
```

Comment:
@cloud-fan Hi Wenchen,

Can you please go through the following comment.

 https://issues.apache.org/jira/browse/SPARK-12988?focusedCommentId=15118433&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15118433

I was trying to address the 3rd bullet in the list. About your second question , per bullet one this should be disallowed ? Please let me know.